src/Templates/MethodTemplate.cshtml

```diff
@@ -24,272 +26,126 @@
 </text>
 }
 
-func (client @(Model.Owner)) @(Model.MethodSignature) (@Model.MethodReturnSignature(false)) {
-@if (Model.ReturnViaChannel)
-{
-<text>
-    resultChan := make(chan @(Model.MethodReturnType), 1)
-    errChan:= make(chan error, 1)
-</text>
-}
-@if ((Model.CodeModel as CodeModelGo).ShouldValidate && !Model.ParameterValidations.IsNullOrEmpty())
+func (client @(Model.Owner)) @(Model.Name)(@Model.MethodParametersSignature(true)) (@Model.MethodReturnSignature(false)) {
+@if ((Model.CodeModel as CodeModelGo).ShouldValidate)
 {
 <text>
-    if err := validation.Validate([]validation.Validation{
+    if err := validate([]validation{
     @(Model.ParameterValidations)}); err != nil {
-    @if (Model.ReturnViaChannel)
-    {
-    <text>
-    errChan @("<-") @(Model.ValidationError)
-    close(errChan)
-    close(resultChan)
-    return resultChan, errChan
-    </text>
-    }
-    else
-    {
-    @:return result, @(Model.ValidationError)
+        return nil, err
     }
-}
-@EmptyLine
-</text>
-}
-@if (Model.ReturnViaChannel)
-{
-<text>
-    go func() {
-        var err error
-        var result @(Model.MethodReturnType)
-
-        defer func() {
-            if err != nil {
-                errChan @("<-") err
-            }
-            resultChan @("<-") result
-            close(resultChan)
-            close(errChan)
-        }()
 </text>
 }
-    req, err := client.@(Model.PreparerMethodName)(@(Model.HelperInvocationParameters(false)))
-    if err != nil {
-        err = @(Model.AutorestError("Failure preparing request"))
-        return
-    }
-@EmptyLine
-    resp, err := client.@(Model.SenderMethodName)(req)
-    if err != nil {
-        @(Model.Response)
-        err = @(Model.AutorestError("Failure sending request", "resp"))
-        return
-    }
-@EmptyLine
-    result, err = client.@(Model.ResponderMethodName)(resp)
+	req, err := client.@(opIdCamelCase)Preparer(@(Model.HelperInvocationParameters(false, false)))
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Pipeline().Do(ctx, responderPolicyFactory{responder: client.@(opIdCamelCase)Responder}, req)
```

Comment:
>ctx [](start = 35, length = 3)

Context might or not be generated?

Reply:
Context will always be the first param to all APIs.